osu.Game/Online/Leaderboards/LeaderboardScore.cs

```diff
@@ -359,5 +367,23 @@ public LeaderboardScoreStatistic(IconUsage icon, string name, string value)
                 Value = value;
             }
         }
+
+        private void deleteLocalScore(ScoreInfo score)
+        {
+            if (score == null || score.ID <= 0) return;
+
+            dialogOverlay?.Push(new LocalScoreDeleteDialog(score));
+        }
+
+        public MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                return (this.allowHighlight) ? null : new MenuItem[]
```

Comment:
This seems like the wrong variable to use here. Maybe you can just use the databased `score.ID` instead?

Reply:
My intention here is to have only the `ContextMenu` show up for local scores, so any online scores will not be affected. 

Please let me know if that is still the wrong choice of variable to use in this context.

Reply:
My point is that allowHighlight doesn't indicate that a score is local or not. Maybe we'll want to allow highlighting local scores in the future, for example, in which case `allowHighlight` will be true for local scores too.

`score.ID` being non-zero should tells you the score was definitely sourced from the database (thus, definitely local).

Reply:
I see what you are saying.

I got the impression that `allowHighlight` is directly related to local from `BeatmapLeaderboard` when it called the `LeaderboardScore`.
```
protected override LeaderboardScore CreateDrawableScore(ScoreInfo model, int index)
{
    model.Beatmap = beatmap;

    return new LeaderboardScore(model, index, IsOnlineScope)
    {
        Action = () => ScoreSelected?.Invoke(model)
    };
}
```

I was unaware that `score.ID` being non-zero basically mean it is local. That being said. I will made the changes accordingly.